pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -74,6 +74,31 @@ func (v *ControlPlaneMutator) Handle(ctx context.Context, req admission.Request)
 		}
 	}
 
+	newOpenShiftRoute := mutator.NewOpenShiftRoute()
+	if newOpenShiftRoute == nil {
+		oldVersion, err := versions.ParseVersion(mutator.OldVersion())
+
+		if err == nil {
+			oldOpenShiftRoute := mutator.OldOpenShiftRoute()
+
+			if oldOpenShiftRoute == nil {
+				switch req.AdmissionRequest.Operation {
+				case admissionv1beta1.Create:
+					mutator.SetOpenShiftRoute(false)
+				case admissionv1beta1.Update:
+					if oldVersion.LessThan(versions.V2_5) {
+						mutator.SetOpenShiftRoute(true)
+					} else {
+						mutator.SetOpenShiftRoute(false)
+					}
```

Comment:
So, once users get this new operator, any update to the SMCP will set openShiftroute to true (for pre-2.5 versions)? Can't we keep it unset? 

Is there a unit test for this code? I'd like to see examples of what this code does and comments on why it does it, as it's not obvious.